zip-liberated-payments.rst

```diff
@@ -0,0 +1,409 @@
+
+Applink_ mechanisms let domain name owners provide a whitelist, specifying which apps are authorized to handle URLs with that domain name. This is implemented by serving suitable files at well-known paths on the web server of that domain or, in the case of a subdomain, its parent domain. Thus, the owner of the ``withzcash.com`` domain effectively controls the whitelist of apps that may be launched by users’ platform to handle URI-Encapsulated Payments (see `Security Implications`_). This whitelist should protect users from installing rogue apps that intercept incoming payments. Thus, the domain owner MUST do the following:
+* Maintain such a whitelist and serve it as needed for the applink_ mechanisms of major platforms.
+* Publish a policy for inclusion of apps in this whitelist.
+* Use all feasible means to whitelist only apps that comply with the published policy.
+* Publish the whitelist’s content in human-readable form.
+* Provide clear and effective means for rapid removal of apps from the whitelist when required as security response.
+* Use all feasible means to protect the whitelist’s integrity (in particular, this includes protecting the web server that serves the whitelist, the domain’s TLS certificate, and the means by which the whitelist is modified).
+* Use effective means for keeping a precise, irrevocable and public history of the whitelist (e.g., using a timestamped Git repository, or an accountability mechanism akin to Certificate Transparency).
+
+They also SHOULD:
+* Strive for the whitelist to include all apps that would not place the user at any greater security risk than reputable state-of-the-art wallet apps.
+
+
+Testing
+-------
+
+For testing purposes, all of the above specification is duplicated for the Zcash testnet network, substituting ``TAZ`` (Zcash testnet coins) for ``ZEC`` and ``testzcash.com`` for ``withzcash.com``.
+
+A separate “testnet whitelist” MUST be maintained by the owner of the ``testzcash.com`` domain name, with a separate policy that SHOULD allow any legitimate third-party developer to add their work-in-progress wallet for testing purposes. Integrity and availability MAY be looser.
+
+Wallets apps MAY support just one type of payments (ZEC or TAZ), and if they support both then they MUST keep separate accounting and must clearly distinguish the type when payments or balances are conveyed to users.
+
+
+Rationale for URI Format
+------------------------
+
+The URI format ``https://pay.withzcash.com:65536/v1#...``  was chosen to allow automatic triggering of wallet software on mobile devices, using the platform’s applink_ mechanism, while minimizing the risk of payment information being intercepted by third parties. The latter is prevented by a defense in depth, where any of the following suffices to prevent the payment information from being exposed over the network:
```

Comment:
```suggestion
The URI format ``https://pay.withzcash.com:65536/v1#...`` was chosen to allow automatic triggering of wallet software on mobile devices, using the platform’s applink_ mechanism, while minimizing the risk of payment information being intercepted by third parties. The latter is prevented by a defense in depth, where any of the following suffices to prevent the payment information from being exposed over the network:
```